modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/DropdownMenu.js

```diff
@@ -24,16 +26,48 @@ export default function DropdownMenu({
 	menuProps,
 	portletId: _portletId,
 	portletNamespace: _portletNamespace,
+	searchable,
 	swapIconSide,
 	...otherProps
 }) {
+	const [filteredItems, setFilteredItems] = useState(items);
+	const [searchValue, setSearchValue] = useState('');
+
+	const searchableProps = searchable
+		? {
+				onSearchValueChange: setSearchValue,
+				searchValue,
+				searchable,
+		  }
+		: {};
+
+	useEffect(() => {
+		if (!searchable) {
+			return;
+		}
+
+		setFilteredItems(() => {
+			return searchValue
+				? items.filter(
+						({label}) =>
+							label
+								.toLowerCase()
+								.indexOf(searchValue.toLowerCase()) !== -1
+				  )
+				: items;
+		});
+	}, [items, searchValue, searchable]);
```

Comment:
Can we implement this without `useEffect`? Something like
```js
onSearchValueChange={(value) => {
		setFilteredItems(() => {
			return value
				? items.filter(
						({label}) =>
							label
								.toLowerCase()
								.indexOf(value.toLowerCase()) !== -1
				  )
				: items;
		});

		setSearchValue(value);
}}
```

Reply:
updated!